### PR TITLE
Add content change diff

### LINKED ISSUE #7: Include text differences in actions and reports
As an action user I want to see the change within the pages of the documentation so that I can check exactly what will be changed as a result of running the action.

Helpful library: https://docs.python.org/3/library/difflib.html

### PR DESCRIPTION
**Pull request by jdkandersson**
Closes #7

### CODE REVIEW

**Comment by jdkandersson** [comment]
The integration tests are failing due to a problem with the checks for the linters, there is a PR here that fixes the `operator`workflows`: https://github.com/canonical/operator-workflows/pull/67

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Review by gregory-schiano — DISMISSED** [review]
LGTM, nice feature

**Comment by jdkandersson** [comment]
Looking at the runner output, I'm not happy with the way the diff is logged yet. Thanks for the reviews, will make some more changes before merging

**Comment by jdkandersson** [comment]
Good to go now

**Review by merkata — APPROVED** [review]
Good addition!

**Review by gregory-schiano — APPROVED** [review]
LGTM

**Review by nrobinaubertin — APPROVED** [review]
LGTM

**Review by arturo-seijas — APPROVED** [review]
LGTM

File: src/action.py [hunk 1]
```diff
@@ -28,6 +29,26 @@ def _absolute_url(url: types_.Url | None, discourse: Discourse) -> types_.Url |
...
+
+    Args:
+        old: The previous content.
+        new: The current content.
+    """
+    old = f"{old}\n" if not old.endswith("\n") else old
+    new = f"{new}\n" if not new.endswith("\n") else new
+    if new != old:
+        logging.info(
+            "content change:\n%s",
```
Comment (weiiwang01) [inline src/action.py:43]: The name of changed files should be added to the log message, to make it more clear.

Comment (jdkandersson) [reply src/action.py:43]: This appears right after the log message with the action (at the start of this function) which includes the file being changed. Do you think that is sufficiently clear?

Comment (jdkandersson) [reply src/action.py:43]: This line logs the action: `logging.info("dry run: %s, action: %s", dry_run, action)` which has the following attributes:

```Python
@dataclasses.dataclass
class UpdateAction:
    """Represents a page to be updated.

    Attrs:
        level: The number of parents, is 1 if there is no parent.
        path: The a unique string identifying the navigation table row.
        navlink_change: The changeto the navlink.
        content_change: The change to the documentation content.
    """

    level: Level
    path: TablePath
    navlink_change: NavlinkChange
    content_change: ContentChange | None
```

**Comment by jdkandersson** [comment]
Needed to merge changes and resolve conflict from main

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Review by yanksyoon — APPROVED** [review]
LGTM!

**Pull request merged**